### PR TITLE
chore: migrate tslint to eslint

### PR DESCRIPTION
**Pull request by GabrielCousin**
## 📝 What is this PR

Taking some time to migrate linting stack to non-depreacted stuff:
- tslint is deprecated since 2019
- run `npx tslint-to-eslint-config` and adapted the config
- remove tslint comments
- update the CI so it at least run lint command

### CODE REVIEW

**Review by val92130 — DISMISSED** [review]
nice, finally 😄 
LGTM

**Review by Hatscat — DISMISSED** [review]
LGTM 💯

File: tsconfig.json [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+	"compilerOptions": {
+		"target": "es5",
+		"module": "commonjs",
+		"moduleResolution": "node",
+		"lib": ["dom", "es2015", "es2016", "es2017", "es2018"],
+		"alwaysStrict": true,
+		"noFallthroughCasesInSwitch": true,
+		"noImplicitAny": true,
+		"noImplicitThis": true,
+		"noEmit": true,
```
Comment (RiahiKarim) [inline tsconfig.json:12]: This new config prevents the web extension from being built.
```suggestion
```

<details><summary>Build error when I execute npm run build: dev</summary>

image


</details>

Comment (GabrielCousin) [reply tsconfig.json]: AHHH good catch! Thanks! It took me a couple of minutes to recall why I put this but it should be fine now 🙌

**Review by RiahiKarim — APPROVED** [review]
LGTM 💯

**Review by Hatscat — APPROVED** [review]

**Review by val92130 — APPROVED** [review]
LGTM

**Pull request merged**